lib/decorators/EventPropertiesDecorator.hpp

```diff
@@ -114,6 +114,9 @@ namespace MAT_NS_BEGIN {
                 record.data.push_back(data);
             }
 
+            if (eventProperties.GetTimestamp() != 0)
```

Comment:
Minor nit: I would've cached in a local variable to avoid calling `GetTimestamp()` twice.

Reply:
e.g.
```cpp
auto ts = eventProperties.GetTimestamp();
if (ts != 0)
  record.time = ts;
```